Bootstrap sigmoid and quadratic

TruckN-Go opened the pull request.

TruckN-Go committed:
sigmoid and quadratic bootstraps added

TruckN-Go committed:
corrected the function name for the numba-fied quadratic bootstrap function

TruckN-Go committed:
fixedd linear regression function

TruckN-Go committed:
fixed so we shuffle each person (not each data point)

TruckN-Go committed:
fixed the tests

SethSullivan reviewed (changes requested):
1. rename quadratic_func to fit_quadratic, or something like that (same for sigmoid_func)
  - Add comment for sigmoid_func. I forget why you're taking the log?
3. Fix comment on line 241.  Also, do you need to pass calc_func? Isn't _nb_bootstrap_quadratic only ever going to use quadratic_func
4. Clean up dead commented code in bootstrap_linear_regression and boostrap_sigmoid
5. Make sure you are only using arguments that are actually used. It doesn't look like return_distribution or stat_type are used in bootstrap_sigmoid
6. Is there supposed to be a boostrap_quadratic function too?

TruckN-Go committed:
fixes from seth's comments pt 1, added numba_fitting.py

1. Linear regression is used to solve the coefficients of a linearized sigmoid in sigmoid_func. linear_regression_func can be accessed as it's own function too to return slope and intercept
2 & 5. We're taking away the quadratic for now. Too many terms to work with. Moved our analytical quadratic solutoin to a new script called numba_fitting.py and made a test script for it too
3 & 4. Just cleaned those up

TruckN-Go committed:
fixed sigmoid and linear regression bootstrapping

TruckN-Go committed:
bootstrap removed from create_mock_data functions

bootstrap removed from create_mock_data functions
